app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +95,50 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_map()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    # pylint: disable=too-many-nested-blocks
```

Comment:
We shouldn't just add a pylint ignores for things like this, we should try to break the method down or separate it out?

Reply:
since this function call comes after `self._answers_by_id` has been set - could you iterate over that instead? then it  could just be 
```
for answer_id, answer in self._answers_by_id.items():
  if (answer_type := ....
````

Reply:
> since this function call comes after `self._answers_by_id` has been set - could you iterate over that instead? then it could just be
> 
> ```
> for answer_id, answer in self._answers_by_id.items():
>   if (answer_type := ....
> ```

This will simplify it a lot.